infrastructure/stacks/api-gateway-sqs/api-gateway.tf

```diff
@@ -27,21 +26,36 @@ resource "aws_api_gateway_method" "di_endpoint_method" {
   api_key_required = true
 }
 
+resource "aws_api_gateway_method_settings" "di_endpoint_method_settings" {
+  rest_api_id = aws_api_gateway_rest_api.di_endpoint.id
+  stage_name  = aws_api_gateway_stage.di_endpoint_stage.stage_name
+  method_path = "*/*"
+
+  settings {
+    metrics_enabled = true
+    logging_level   = "INFO"
+  }
+}
+
 resource "aws_api_gateway_integration" "di_endpoint_integration" {
   http_method             = aws_api_gateway_method.di_endpoint_method.http_method
   resource_id             = aws_api_gateway_resource.di_endpoint_change_event_path.id
   rest_api_id             = aws_api_gateway_rest_api.di_endpoint.id
   integration_http_method = "POST"
   type                    = "AWS"
-  uri                     = "arn:aws:apigateway:${var.aws_region}:sqs:path/${var.fifo_queue_name}"
+  uri                     = "arn:aws:apigateway:${var.aws_region}:sqs:path/${var.aws_account_id}/${var.fifo_queue_name}"
   credentials             = aws_iam_role.di_endpoint_role.arn
   passthrough_behavior    = "NEVER"
+  request_parameters = {
+    "integration.request.header.Content-Type" = "'application/x-www-form-urlencoded'"
+  }
   request_templates = {
-    "application/json" = "Action=SendMessage&MessageBody=$input.body"
+    "application/json" = "Action=SendMessage&MessageGroupId=$input.json(\"ODSCode\")&MessageBody=$input.body"
```

Comment:
Can we also send the correlation-id as a message attribute, i.e. map the headers.x-correlation-id to a message attribute ... https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SendMessage.html#API_SendMessage_Examples

Reply:
pretty much although googling around suggests you need $input.params().header.get to pull from incoming http headers
`MessageAttribute.1.Value.StringValue=$util.escapeJavaScript($input.params().header.get('x-correlation-id'))`

Reply:
The `$util.escapeJavaScript()` is recommended to stop potential injection attacks though the correlation id

Reply:
Do you want correlation id as `x-correlation-id'` or `correlation_id` as the latter is used in the event processor. Should this be changed?

Reply:
hmm can probably drop the `x-` https://www.keycdn.com/support/custom-http-headers#naming-conventions, would keep it spinal-case as this seems to be general custom for http headers, now that doesnt mean it has to be like that when it gets to the event processor as that is not a http interface, the code will need to be updated anyway to pull the correlation id from the SQS message attributes, this could be done in a middleware